cpp/arcticdb/pipeline/read_frame.cpp

```diff
@@ -1013,7 +1013,7 @@ std::unique_ptr<StringReducer> get_string_reducer(
             const auto alloc_width = get_max_string_size_in_column(column.data().buffer(), context, frame, frame_field, slice_map, true);
             string_reducer = std::make_unique<UnicodeConvertingStringReducer>(column, context, frame, frame_field, alloc_width);
         } else {
-            const auto alloc_width = get_max_string_size_in_column(column.data().buffer(), context, frame, frame_field, slice_map, false);
+            const auto alloc_width = get_max_string_size_in_column(column.data().buffer(), context, frame, frame_field, slice_map, true);
```

Comment:
This will be prohibitively expensive for large dataframes with fixed-width strings that have not been aggregated, and so `false` would be sufficient. You need to pass the information in to this method of what this parameter should be.

Reply:
A hacky solution would be to set the orig_type in the aggregator/clause, so that was_coerced_from_dynamic_to_fixed is true in this case. Then add a comment that this is a big hack. Long term we want all reads to go through the `ComponentManager`, at which point it will decide what to do, so this is OK for now.

Reply:
I first tried the hacky solution, but it's just bringing more complexity since the final dataframe to be compared with the expected one in the tests turns out to have different indexes and data regarding encoding, which involves additional hacking to do.
I think the first solution is fair enough (see this [commit](https://github.com/man-group/ArcticDB/pull/1151/commits/54a190e6a6f765ff7b88ecc0fda44bee015ddfd2#diff-b0ecf8c7fd4b7a415c9bb858bef66da846d90d1c7499a18057562c1375200972)).